src/RuleSet/RuleSetInterface.php

```diff
@@ -42,6 +42,17 @@ public function getRuleConfiguration(string $rule): ?array;
      */
     public function getRules(): array;
 
+    /**
+     * Get information about rules' source. Each rule can be defined on several levels:
```

Comment:
I don't think it is a RuleSets responsibility to known, keep track or provide information about how it was created. I think it might be more fitting in what ever component resolves the configuration to keep track of that sorts of information.

Reply:
You would be right, if it wasn't the ruleset itself who resolves final state of the rules 🤷‍♂️. According to how it is done currently I don't see a value in resolving subsets and unpacking rules inside ruleset, and storing info about sources externally, because this should be in the same place. So either we need this method here, or we need to move resolving logic somewhere else and keep ruleset container dumb and provide this information from the actual resolver (which currently proxies it from the ruleset).